HeterogeneousCore/AlpakaCore/src/alpaka/chooseDevice.cc

```diff
@@ -0,0 +1,41 @@
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/stringize.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/chooseDevice.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/devices.h"
+#include "HeterogeneousCore/AlpakaServices/interface/alpaka/AlpakaService.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::detail {
+  Device const& chooseDevice(edm::StreamID id) {
+    // The idea of checking the AlpakaService status here is that
+    // regardless of process.options.accelerators setting the
+    // configuration may contain an EDModule for a specific Alpaka
+    // backend. Checking the AlpakaService status here ensures that an
+    // explanatory error messages even in that case. The information
+    // on the intended accelerators is (eventually) communicated from
+    // process.options.accelerators to AlpakaServices via
+    // ProcessAcceleratorAlpaka.
+    edm::Service<ALPAKA_ACCELERATOR_NAMESPACE::AlpakaService> alpakaService;
+    if (not alpakaService->enabled()) {
```

Comment:
I decided to duplicate the `chooseDevice()` from `cms::alpakatools::chooseDevice()` in order to avoid the templating, add easily the dependence on the corresponding `AlpakaService` (following the discussion in https://github.com/cms-sw/cmssw/pull/39319#discussion_r963902507).